cmd/tsbs_load_mongo/creator.go

```diff
@@ -61,36 +61,57 @@ func (d *dbCreator) CreateDB(dbName string) error {
 	}
 
 	res := d.client.Database(dbName).RunCommand(context.Background(), cmd)
+
 	if res.Err() != nil {
 		if strings.Contains(res.Err().Error(), "already exists") {
 			return nil
 		}
 		return fmt.Errorf("create collection err: %v", res.Err().Error())
 	}
 
-	if timeseriesCollection && timeseriesCollectionSharded {
-		// first enable sharding on dbName
+	if collectionSharded {
+	        // first enable sharding on dbName
 		cmd1 := make(bson.D, 0, 4)
 		cmd1 = append(cmd1, bson.E{"enableSharding", dbName})
-
-		res1 := d.client.Database("admin").RunCommand(context.Background(), cmd1)
-		if res1.Err() != nil {
+		
+	        res1 := d.client.Database("admin").RunCommand(context.Background(), cmd1)
+	        if res1.Err() != nil {
 			return fmt.Errorf("enableSharding err: %v", res1.Err().Error())
 		}
 
-		// then shard the collection on timeField
+		// then shard the collection
 		cmd2 := make(bson.D, 0, 4)
-		cmd2 = append(cmd2, bson.E{"shardCollection", dbName + "." + collectionName})
-		cmd2 = append(cmd2, bson.E{"key", bson.M{timestampField: 1}})
-
-		res2 := d.client.Database("admin").RunCommand(context.Background(), cmd2)
+		cmd2 = append(cmd2, bson.E{"shardCollection",dbName+"."+collectionName})
+		var shardKey interface{}
+		
+		err := bson.UnmarshalExtJSON([]byte(shardKeySpec), true, &shardKey)
+		if err != nil {
+		   err = bson.UnmarshalExtJSON([]byte("{\"time\":1}"), true, &shardKey)		       
+		}
+		cmd2 = append(cmd2, bson.E{"key", shardKey})
 
+		if numInitChunks > 0 {
+		   	cmd2 = append(cmd2, bson.E{"numInitialChunks", numInitChunks})
+		}
+	   	res2 := d.client.Database("admin").RunCommand(context.Background(), cmd2)
+	
 		if res2.Err() != nil {
-			return fmt.Errorf("shard collection err: %v", res2.Err().Error())
+		        return fmt.Errorf("shard collection err: %v", res2.Err().Error())
+	        }
+
+		cmd3 := make(bson.D, 0, 4)
```

Comment:
Can you rename `cmd1`, `cmd2` and `cmd3` with `enableShardingCmd`, `shardCollCmd` and `balancerCmd` to make the code a bit more readable?

Reply:
done.